proposals/NNNN-explicit-padding.md

```diff
@@ -0,0 +1,175 @@
+---
+title: "[NNNN] - Explicit Padding in Structs and CBuffer Arrays"
+params:
+    status: Design In Progress
+    authors:
+        - bogner: Justin Bogner
+---
+
+
+## Introduction
+
+We introduce an explicit padding type for HLSL, and construct cbuffer arrays
+and structs that are annotated with `packoffset` or `vk::offset` using this
+type to unambiguously lay out these objects.
```

Comment:
```suggestion
We introduce an explicit padding type for HLSL, and construct cbuffer arrays,
structs, and cbuffer elements that are annotated with `packoffset` or `vk::offset` using this
type to unambiguously lay out these objects.
```

Reply:
Structs in cbuffer must also start on a 16-byte boundary and we need to insert padding before if needed: https://godbolt.org/z/YEjc7TEY9